HabitTracker.ConsoleApp-DreamFXX/Program.cs

```diff
@@ -0,0 +1,366 @@
+                    break;
+                case "1":
+                    ViewAllRecords();
+                    break;
+                case "2":
+                    AddRecord();
+                    break;
+                case "3":
+                    DeleteRecord();
+                    break;
+                case "4":
+                    ChangeRecord();
+                    break;
+                case "5":
+                    AddNewHabit();
+                    break;
+                default:
+                    Console.WriteLine("\n\nInvalid number of operation. Try again, valid operations are in range 0 - 5.\n\n");
+                    break;
+            }
+        }
+    }
+
+    private static void ViewAllRecords()
+    {
+        Console.Clear();
+
+        using (var connection = new SQLiteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText = "SELECT * FROM HabitRecords";
+
+            List<HabitRecord> tableData = new List<HabitRecord>();
+            SQLiteDataReader reader = tableCmd.ExecuteReader();
+
+            if (reader.HasRows)
+            {
+                while (reader.Read())
+                {
+                    tableData.Add(
+                        new HabitRecord
+                        {
+                            Id = reader.GetInt32(0),
+                            HabitId = reader.GetInt32(1),
+                            Date = reader.GetString(2),
+                            Time = reader.GetString(3),
+                            Quantity = reader.GetInt32(4),
+                        });
+                }
+            }
+            else
+            {
+                Console.WriteLine("\n\nNo records were found!\n\n");
+            }
+
+            connection.Close();
+
+            Console.WriteLine("------------HABIT RECORDS-----------\n");
+            foreach (var record in tableData)
+            {
+                Console.WriteLine($"{record.Id} -> {record.Date} in {record.Time}h // {record.Quantity}");
+            }
+            Console.WriteLine("------------------------------------\n");
+        }
+    }
+
+    private static void AddRecord()
+    {
+        Console.WriteLine("Choose a habit by entering Habits ID number below!");
+        ViewHabits();
+
+        int habitId = GetNumberInput("Habit ID:");
+        string date = GetDate();
+        string time = GetTime();
+        int quantity = GetNumberInput("Enter quantity of habit you consumed // ran // did in units you selected in specified habit tracking.");
+
+        using (var connection = new SQLiteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText = $"INSERT INTO HabitRecords (HabitId, Date, Time, Quantity) VALUES ({habitId}, '{date}', '{time}', {quantity})";
```

Comment:
🟠 Parameterised SQL Queries

💡 I know this is technically a challenge requirement. But every query should be parameterised as it is not hard to refactor and the dangers of not doing so are great.

🔧 
❌
```
tableCmd.CommandText = $"INSERT INTO HabitRecords (HabitId, Date, Time, Quantity) VALUES ({habitId}, '{date}', '{time}', {quantity})";
```

✔️ 
```
tableCmd.CommandText = $"INSERT INTO HabitRecords (HabitId, Date, Time, Quantity) VALUES (@HabitId, @Date, @Time, @Quantity)";
command.Parameters.AddWithValue("@HabitId", habitId);
command.Parameters.AddWithValue("@Date", date);
command.Parameters.AddWithValue("@Time", time);
command.Parameters.AddWithValue("@Quantity", quantity);
```